src/NuGet.Indexing/StorageLoader.cs

```diff
@@ -7,21 +7,34 @@
 using Newtonsoft.Json;
 using System;
 using System.IO;
+using System.Threading.Tasks;
 using FrameworkLogger = Microsoft.Extensions.Logging.ILogger;
 
 namespace NuGet.Indexing
 {
     public class StorageLoader : ILoader
     {
-        private readonly CloudStorageAccount _storageAccount;
         private readonly string _containerName;
         private readonly FrameworkLogger _logger;
 
-        public StorageLoader(CloudStorageAccount storageAccount, string containerName, FrameworkLogger logger)
+        private CloudStorageAccount _storageAccount;
+        private AzureDirectorySynchronizer _synchronizer;
+        private Func<Task<Tuple<CloudStorageAccount, AzureDirectorySynchronizer>>> _createCloud;
+
+        public static async Task<StorageLoader> CreateStorageLoader(Func<Task<Tuple<CloudStorageAccount, AzureDirectorySynchronizer>>> createCloud, string containerName, FrameworkLogger logger)
```

Comment:
I think passing in parameters sounds like a much better implementation. I will fix.